### ISSUE TITLE
Eglot Bash Language Server

### ISSUE THREAD

**Issue by iris-garcia** [joaotavora/eglot#597]
Emacs becomes almost unusable, I can see this error messages:

```
error in process filter: eglot--range-region: Wrong type argument: number-or-marker-p, nil
error in process filter: Wrong type argument: number-or-marker-p, nil
error in process filter: eglot--range-region: Wrong type argument: number-or-marker-p, nil
error in process filter: Wrong type argument: number-or-marker-p, nil
```
```
(:jsonrpc "2.0" :method "window/logMessage" :params
          (:type 4 :message "onDocumentHighlight 83:0 "))
[server-reply] (id:21) Thu Jan 21 10:07:19 2021:
(:jsonrpc "2.0" :id 21 :result
          [(:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
           (:range nil)
```

**Comment by joaotavora** [comment]
Hello, thanks for reporting this.  It seems the server is replying to _something_ with invalid results and Eglot isn't being robust enough to handle that.  So this is definitely a bug in Eglot _and_ in the bash language server.  But it would be nice to have more of that transcript...

**Comment by joaotavora** [comment]
Presumably easy to fix on the Eglot side, so adding the "help wanted" label, as I don't have much time right now.

**Comment by iris-garcia** [comment]
I'll try to debug this, but my elisp knowledge is very limited, lets see how far I get.

**Comment by joaotavora** [comment]
For starts, `M-x toggle-debug-on-error` will help a lot

**Comment by iris-garcia** [comment]
After a day debugging this, I found out I didn't have the latest version of the Bash Language Server, updating to the latest version seems to fix this issue, I guess we can close the issue for now, sorry :(.

**Comment by joaotavora** [comment]
Yes, I don't mean to belittle your effort (one full day, I'm so sorry :-) ), but it was clear that this is a server problem.  However, Eglot should behave more responsibly when given this invalid data, and this is why I am reopening.

**Comment by iris-garcia** [comment]
No worries, I really want to get into elisp, I just find it hard to learn (coming from OOP). But I'll install again the old Bash Language Server and try to handle this invalid data in a better way.

**Comment by skangas** [comment]
@iris-garcia Did you get anywhere with this?

Could you tell us which version of the bash language server caused this?
